src/string-builder.js

```diff
@@ -52,4 +67,34 @@ StringBuilder.prototype.catIf = function(...args) {
     }
 
     return this;
+};
+
+StringBuilder.prototype.wrap = function(prefix, suffix) {
+    this.decorators.push('wrap');
+    if (prefix) {
+        this.prefix = prefix;
+    }
+
+    if (suffix) {
+        this.suffix = suffix;
+    }
+
+    return this;        
+};
+
+StringBuilder.prototype.end = function(deep) {   
+    if(deep) {
+        let aux = this.decorators.pop();
+        //This is part is going to change with the implementation of prefix and suffix
+        if(aux && aux === 'wrap') {
```

Comment:
This kind of guards are normally (encouraged to be) used when you want to test an *object*'s inner property, such way you need to prevent **null exceptions** e.g.: `if(obj && obj.prop === 'foo')`

In other words, you only need `if(aux === 'wrap')`. In case of `aux` being `null` or `undefined` it wont throw any `exception` but the expression will evaluate to `false`.